src/app/dataset/depublication/depublication.component.ts

```diff
@@ -323,24 +341,28 @@ export class DepublicationComponent extends DataPollingComponent implements OnDe
   /* - handler for depublish record ids button
   /* - invoke service call
   /* - flag success / trigger reload / clear selection cache
+  /*  @param {boolean} all - optional - flag to send empty (all) or selected
   */
-  onDepublishRecordIds(): void {
-    if (this.depublicationSelections.length === 0) {
-      return;
-    }
+  onDepublishRecordIds(all?: boolean): void {
     this.isSaving = true;
-    this.depublications.depublishRecordIds(this._datasetId, this.depublicationSelections).subscribe(
-      () => {
-        this.depublicationSelections = [];
-        this.pollingRefresh.next(true);
-        this.isSaving = false;
-      },
-      (err: HttpErrorResponse): void => {
-        this.onError(err);
-      }
-    );
+    this.depublications
+      .depublishRecordIds(this._datasetId, all ? [] : this.depublicationSelections)
```

Comment:
I know it is not needed (because the button would be disabled) but perhaps we could do a sanity check here that if `all` is false but `this.depublicationSelections` is empty, it doesn't send an empty list to the server thus causing all records to be deleted by accident?

Reply:
Added the check